libs/utils/src/vec_map.rs

```diff
@@ -147,6 +147,17 @@ impl<K: Ord, V> VecMap<K, V> {
     }
 }
 
+impl<K: Ord, V> FromIterator<(K, V)> for VecMap<K, V> {
+    fn from_iter<I: IntoIterator<Item = (K, V)>>(iter: I) -> Self {
+        let mut vec_map = VecMap::default();
+
+        vec_map.0 = iter.into_iter().collect();
+        vec_map.0.sort_by(|(k1, _), (k2, _)| k1.cmp(k2));
+
+        vec_map
+    }
+}
```

Comment:
You're still sorting here. This should assert out if `iter` is not sorted.

I'd do something like:
```
impl<K: Ord, V> FromIterator<(K, V)> for VecMap<K, V> {
    fn from_iter<I: IntoIterator<Item = (K, V)>>(iter: I) -> Self {
        let initial_capacity = {
          match iter.size_hint() {
            (lower_bound, None) => lower_bound,
            (_lower_bound, Some(upper_bound)) => upper_bound
          }
        };
        
        // You'll have to add a `with_capacity` constructor.
        let mut vec_map = VecMap::with_capacity(initial_capacity);
        
        for (key, value) in iter {
          vec_map.append(key, value);
        }

        vec_map
    }
}
```